worker/common/credentialinvalidator.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common
+
+import (
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/api/credentialvalidator"
+	"github.com/juju/juju/environs/context"
+)
+
+// CredentialAPI exposes functionality of the credential validator API facade to a worker.
+type CredentialAPI interface {
+	InvalidateModelCredential(reason string) error
+}
+
+// NewCredentialInvalidatorFacade creates an API facade capable of invalidating credential.
+func NewCredentialInvalidatorFacade(apiCaller base.APICaller) (CredentialAPI, error) {
+	f := credentialvalidator.NewFacade(apiCaller)
+	if f == nil {
+		return nil, errors.New("could not create credential validator facade")
```

Comment:
NewFacade should never return nil without a corresponding error.
I don't know of any facade constructor that returns an error. Do we need the == nil check at all?

Reply:
A few facade constructors return errors - api.agent, ..., api.undertaker... However, it is also true that we have a few that do not and in this case in particular, it's redundant :) I'll remove the check and the error.